src/middlewared/middlewared/plugins/virt/instance_device.py

```diff
@@ -353,6 +354,11 @@ async def validate_device(
                             verrors.add(
                                 schema, 'Source must be a path starting with /dev/zvol/ for VM or a virt volume name.'
                             )
+                elif device['name'] == 'root':
+                    if source != '':
+                        verrors.add(schema, 'Root disk source must be unset.')
+
+                    device['storage_pool'] = storage_pool_to_incus_pool(old['storage_pool'])
```

Comment:
We should be setting `device['storage_pool'] = old['storage_pool']` here then do the conversion to incus when we call device_to_incus().